### PR TITLE
handle task artifacts - workflow executer

### PR DESCRIPTION
**Pull request by jackschofield23**
Changes required for #52 .

### Description

Add task artifacts logic to the workflow executer (workflow creation message and task update message)

### Status
**In Progress**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog
- [x] Any new files have copyright headers
- [x] Code coverage above 70%
- [x] TODO items resolved

### CODE REVIEW

File: src/Common/Extensions/StorageListExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿using Ardalis.GuardClauses;
```
Comment (mocsharp) [inline src/Common/Extensions/StorageListExtensions.cs:1]: copyright

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 2]
```diff
@@ -73,7 +81,7 @@ public async Task<bool> ProcessPayload(WorkflowRequestEvent message)
 
             var workflowInstances = new List<WorkflowInstance>();
 
-            workflows.ForEach((workflow) => workflowInstances.Add(CreateWorkFlowIntsance(message, workflow)));
+            workflows.ForEach(async (workflow) => workflowInstances.Add(await CreateWorkflowInstance(message, workflow)));
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:84]: CreateWorkflowInstance
Should maybe be called CreateWorkflowInstanceAsync now?

Comment (jackschofield23) [reply src/WorkflowExecuter/Services/WorkflowExecuterService.cs]: Good shout, have changed it to that.

File: src/Common/Extensions/StorageListExtensions.cs (hunk 1 above)
Comment (jackschofield23) [reply src/Common/Extensions/StorageListExtensions.cs]: Fixed

File: src/Database/WorkflowInstanceRepository.cs [hunk 3]
```diff
@@ -95,6 +95,26 @@ public async Task<bool> UpdateTaskStatusAsync(string workflowInstanceId, string
...
 
+        public async Task<bool> UpdateTaskOutputArtifactsAsync(string workflowInstanceId, string taskId, Dictionary<string, string> outputArtifacts)
+        {
+            Guard.Against.NullOrWhiteSpace(workflowInstanceId, nameof(workflowInstanceId));
+            Guard.Against.NullOrWhiteSpace(taskId, nameof(taskId));
+            Guard.Against.Null(outputArtifacts, nameof(outputArtifacts));
+
+            try
+            {
+                var result = await _workflowInstanceCollection.FindOneAndUpdateAsync(
```
Comment (mocsharp) [inline src/Database/WorkflowInstanceRepository.cs:106]: if `result` is not used, you may assign it to `_`

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 4]
```diff
@@ -92,6 +100,11 @@ public async Task<bool> ProcessPayload(WorkflowRequestEvent message)
 
             foreach (var workflowInstance in workflowInstances)
             {
+                if (workflowInstance.Status == Status.Failed)
+                {
+                    continue;
+                }
+
                 var task = workflowInstance.Tasks.FirstOrDefault();
```
Comment (mocsharp) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:108]: null check?

**Review by mocsharp — APPROVED** [review]
May want to fix some of the warnings if possible.

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs (hunk 4 above)
Comment (jackschofield23) [reply src/WorkflowExecuter/Services/WorkflowExecuterService.cs:108]: added

File: src/Database/WorkflowInstanceRepository.cs (hunk 3 above)
Comment (jackschofield23) [reply src/Database/WorkflowInstanceRepository.cs]: removed variable

**Pull request merged**